app/views/contexts/section_summary_context.py

```diff
@@ -137,46 +149,53 @@ def _list_summary_element(self, summary, current_location, section) -> Mapping:
                 "id"
             ]
 
-        add_link = self._add_link(
-            summary, current_location, section, list_collector_block
-        )
-
-        list_context = ListContext(
-            self._language,
-            self._schema,
-            self._answer_store,
-            self._list_store,
-            self._progress_store,
-            self._metadata,
-        )
+        add_link = self._add_link(summary, section, list_collector_block)
 
         rendered_summary = self._placeholder_renderer.render(
             summary, current_location.list_item_id
         )
 
+        if list_collector_blocks_on_path:
```

Comment:
This is the 2nd time `list_collector_blocks_on_path` is checked for it's truthiness value. An alternative would be to move the instantions:

```
edit_block_id = None
remove_block_id = None
```

to the start of the method, then override them inside the if block on line 138

Reply:
I moved it up and there is only one if statement based on `list_collector_blocks_on_path` now, if only I could compact it more😆.